scripts/unassign-inactive-issues.mjs

```diff
@@ -0,0 +1,169 @@
+    //   console.log('Issue data:', {
+    //     number: issue.number,
+    //     title: issue.title,
+    //     assignee: assignee.login,
+    //     has_pr_field: !!issue.pull_request,
+    //     updated_at: issue.updated_at
+    //   });
+
+      // Check for linked PRs using multiple methods
+      const checkLinkedPRs = async () => {
+        try {
+          // Method 1: Check direct PR field
+          if (issue.pull_request) {
+            console.log(`Issue #${issue.number} is a pull request`);
+            return true;
+          }
+
+          // Method 2: Check timeline events
+          const timeline = await octokit.rest.issues.listEventsForTimeline({
+            owner,
+            repo,
+            issue_number: issue.number,
+            per_page: 100
+          });
+
+          console.log(`Timeline events for issue #${issue.number}:`, 
+            timeline.data.map(event => ({
+              event: event.event,
+              type: event.event_type,
+              actor: event.actor?.login,
+              created_at: event.created_at
+            }))
+          );
+
+          // Look for various PR-related events
+          const hasPRLink = timeline.data.some(event => 
+            event.event === 'cross-referenced' ||
+            event.event === 'connected' ||
+            event.event === 'referenced' ||
+            (event.source && event.source.type === 'pull_request')
+          );
+
+          if (hasPRLink) {
+            console.log(`Issue #${issue.number} has PR links in timeline`);
+            return true;
+          }
+
+          // Method 3: Check issue body for PR links
+          const prLinkRegex = /(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s+#\d+/i;
+          if (issue.body && prLinkRegex.test(issue.body)) {
+            console.log(`Issue #${issue.number} has PR references in body`);
+            return true;
+          }
+
+          console.log(`No PR links found for issue #${issue.number}`);
+          return false;
+        } catch (error) {
+          console.error(`Error checking PR links for issue #${issue.number}:`, error);
+          return false;
+        }
+      };
+
+      const lastActivity = new Date(issue.updated_at);
+      const now = new Date();
+      
+      if (now - lastActivity > inactivityPeriodInMinutes * 60 * 1000) {
+        const hasLinkedPR = await checkLinkedPRs();
+        
+        if (hasLinkedPR) {
+          console.log(`Skipping issue #${issue.number} as it has linked PR(s)`);
+          continue;
+        }
+
+        console.log(`Processing inactive issue #${issue.number} with no linked PRs`);
+        
+        try {
+          // Unassign user
+          await octokit.rest.issues.update({
+            owner,
+            repo,
+            issue_number: issue.number,
+            assignees: [],
+          });
+
+          console.log(`Successfully unassigned user from issue #${issue.number}`);
+
```

Comment:
@rtibbles I didn't really get yet to think more about tracking assigned issues, but I thought we could at least start with sending Slack notification to #dev-support in case this action unassigns someone here? To have at least some record and an option to manually tweak in case of some issues.

@SukhvirKooner if @rtibbles agrees with this, it would be optional so only if you'd be interested. I could also create a new issue.

Reply:
Hi @MisRob , I agree with the idea and would be happy to work on it under your guidance. Let me know how to proceed or if a new issue is created.

Reply:
Thank you, let me confirm with the team

Reply:
Hi @MisRob ,

Thanks for confirming! I’ll add this functionality to the PR. My approach will be:

- I'll first test the Slack notification in my own repo to ensure it works as expected.

- The notification will use the existing SLACK_WEBHOOK_URL to send a message to the dedicated channel.

- The message will follow the format: `Automatically unassigned '<user_gh_handle>' from <repo_name>#<issue_number>`, where `<issue_number>` will include a clickable link to the issue.

- I’ll handle the notification logic in JavaScript to make it reusable for future Slack integrations.

Once I’ve verified the functionality, I’ll update the PR accordingly. If I need any help , I’ll reach out to you.

Thanks again for the detailed guidance—it’s really helpful!

Reply:
Hi @MisRob ,
I’ll make sure all unassignments are sent in one message by collecting them into a list, formatting them with Slack link syntax, and sending a single notification via the webhook URL to keep the Slack channel organised.

Reply:
Thanks @SukhvirKooner